### PR TITLE
Fixes for clusterroles and bindings and support for PSP

### PR DESCRIPTION
**Pull request by amdabhad**
Fixes for following issues mentoined in defect https://github.com/IBM/ibm-spectrum-scale-csi/issues/536:
- Clusterroles name are not same
- Clusterroles rules are not same i.e. attacher clusterrole
- csi psp for k8s/OCP env not working
- Clusterrolebinding name are not same
- Clusterrole and clusterrolebindig for node plugin is not present

### CODE REVIEW

File: api/v1/csiscaleoperator_types.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ type CSIScaleOperatorSpec struct {
 	// +kubebuilder:default:=/var/lib/kubelet
 	// kubeletRootDirPath is the path for kubelet root directory.
 	KubeletRootDirPath string `json:"kubeletRootDirPath,omitempty"`
+
+	// PodSecurityPolicy name for CSI driver and sidecar pods
+	CSIpspname string `json:"csipspname,omitempty"`
```
Comment (nitishkumar4) [inline api/v1/csiscaleoperator_types.go:111]: Can we rename the spec in CR to something like `CsiPSPName`? like in line 108 for `kubeletRootDirPath`. The reason being, when user describes the custom resource the `csipspname` would look like `Csipspname`, but `kubeletRootDirPath` looks like `Kublet Root Dir Path` which has better readability.

**Review by nitishkumar4 — APPROVED** [review]
LGTM. Have shared a review comment. Please handle it.

File: api/v1/csiscaleoperator_types.go (hunk 1 above)
Comment (amdabhad) [reply api/v1/csiscaleoperator_types.go:111]: I have used the name to be consistent with current CNSA PSP variable: https://github.ibm.com/IBMSpectrumScale/scale-core/blob/master/apis/csi/v1/csi_types.go#L108

File: controllers/csiscaleoperator_controller.go [hunk 2]
```diff
@@ -912,8 +916,18 @@ func (r *CSIScaleOperatorReconciler) updateStatus(instance *csiscaleoperator.CSI
...
+	logger := csiLog.WithName("isControllerReady")
+	logMessage := "Controller status"
+	logKey := "ReadyReplicas == MinControllerReplicas"
+	logValueTrue := "True"
+	logValueFalse := "False"
+	if controller.Status.ReadyReplicas == MinControllerReplicas {
+		logger.Info(logMessage+":", logKey+":", logValueTrue)
+		return true
+	} else {
+		logger.Info(logMessage+":", logKey+":", logValueFalse)
```
Comment (madhuthorat) [inline controllers/csiscaleoperator_controller.go:929]: We can log value of controller.Status.ReadyReplicas here.

**Review by madhuthorat — APPROVED** [review]

File: controllers/csiscaleoperator_controller.go (hunk 2 above)
Comment (amdabhad) [reply controllers/csiscaleoperator_controller.go:929]: Done, added a log statement here.

**Pull request merged**